tests/filters/test_filters.py

```diff
@@ -253,6 +273,30 @@ def fruits(self, filters: FruitFilter) -> List[Fruit]:
     ]
 
 
+def test_resolver_filter_with_inheritance(vegetables):
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def vegetables(self, filters: VegetableFilter) -> List[Vegetable]:
+            queryset = models.Vegetable.objects.all()
+            return cast(
+                List[Vegetable], strawberry_django.filters.apply(filters, queryset)
+            )
+
+    query = utils.generate_query(Query)
+    result = query(
+        "{ vegetables(filters:  { worldProduction: { gt: 100e6 } "
+        'OR: { name: { exact: "cucumber" } } }) { id name } }'
```

Comment:
nit: can we have this formatted properly using a multiline string? check `tests/test_optimizer.py` for some examples